apps/calendar/test/marionette/lib/views/week.js

```diff
@@ -77,6 +81,153 @@ Week.prototype = {
     });
   },
 
+  /**
+   * This method tries to find out where the hour square is located in the
+   * window.
+   *
+   * @param {Number} hour The hour to locate.
+   * @returns {Number} 0 if it's displayed, or the delta otherwise: negative
+   * if it's above or positive if it's below.
+   */
+  _getDeltaToFullyDisplayHour: function(hour) {
+    var hourElt = this.hours[hour];
+    var mainScroll = this.scrollTop;
+    var hourOffset = hourElt.scriptWith(function(el) { return el.offsetTop; });
+    var mainClientHeight = this.main.rect().height;
+    var hourHeight = hourElt.rect().height;
+
+    if (hourOffset < mainScroll) {
+      return hourOffset - mainScroll;
+    }
+
+    var bottomHourOffset = hourOffset + hourHeight;
+    var bottomScroll = mainScroll + mainClientHeight;
+    if (bottomHourOffset > bottomScroll) {
+      return bottomHourOffset - bottomScroll;
+    }
+    return 0;
+  },
+
+  /**
+   * This method waits until we don't scroll.
+   *
+   * Note this doesn't seem to work properly when the view is just displayed.
+   * That's why waitForHourScrollEnd is still useful.
+   */
+  waitForNoScroll: function() {
```

Comment:
I feel it could be used in a public way though. But I'd prefer to see something like this, and working in all conditions, in a generic helper.